### DISCUSSION TITLE
Setting a minimum time at the trial level

### DISCUSSION THREAD

**Discussion by arthurprat** [jspsych/jsPsych#3260] [Q&A] [answered]
In a `html-keyboard-response` trial, I would like to set a minimum duration (e.g., 1 second) before allowing my subjects to provide a response. The parameter `minimum_valid_rt` does this job but for the whole experiment (it's an argument to `jsPsych.init`), which is not what I want.

The parameters of an `html-keyboard-response` do not seem to do what I want (e.g., `trial_duration` sets the maximal duration if `response_ends_trial` is `false`, and it sets the _maximal_ duration if `response_ends_trial` is `true`.) 

Is there any straightforward way to do this? In #886 the following solution is given by @vijaymarupudi : make two trials, with the first one displayed for the duration I want and not accepting any response, and the second one recording a response. I am wondering if there is a better way.

**Comment by Shaobin-Jiang** [comment, marked as answer]
You can just:

- set the `choices` to something probably no one would have on their keyboard
- set up your own keyboard listener in `on_load` with a `minimum_valid_rt`
- feed the `choices` key in the callback

Example given below:

```javascript
let jsPsych = initJsPsych();

let trial = {
    type: jsPsychHtmlKeyboardResponse,
    stimulus: 'Minimum rt: 3000 ms',
    choices: ['Symbol'],
    on_load: function () {
        jsPsych.pluginAPI.getKeyboardResponse({
            callback_function: function () {
                jsPsych.pluginAPI.keyDown('Symbol');
            },
            valid_responses: 'ALL_KEYS',
            minimum_valid_rt: 3000,
        });
    }
};

jsPsych.run([trial]);
```

You can, of course, use a much more secure way by setting `choices` to `NO_KEYS` and deal with keyboard responses entirely by yourself, but that would involve data recording as well, which is why I prefer the solution above.

**Reply by arthurprat** [reply]
Sorry I'm replying only now -- thank you for your great answer!
